sklearn/ensemble/stacking_classifier.py

```diff
@@ -0,0 +1,295 @@
+          - integer>1, to specify the number of folds in a `(Stratified)KFold`,
+          - An object to be used as a cross-validation generator.
+          - An iterable yielding train, test splits.
+
+        For integer/None inputs, if the estimator is a classifier and ``y`` is
+        either binary or multiclass, :class:`StratifiedKFold` is used. In all
+        other cases, :class:`KFold` is used.
+
+        Refer :ref:`User Guide <cross_validation>` for the various
+        cross-validation strategies that can be used here.
+
+    verbose : integer
+        Controls the verbosity: the higher, the more messages.
+
+    n_jobs : int, optional (default=1)
+        The number of jobs to run in parallel for ``fit``.
+        If -1, then the number of jobs is set to the number of cores.
+
+    Attributes
+    ----------
+    estimators_ : list of classifiers
+        The collection of fitted sub-estimators.
+
+    meta_estimators_ : classifier
+        Fitted meta-estimator
+
+    classes_ : array-like, shape = [n_predictions]
+        The classes labels.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> from sklearn.naive_bayes import GaussianNB
+    >>> from sklearn.ensemble import RandomForestClassifier, StackingClassifier
+    >>> clf1 = LogisticRegression(random_state=1)
+    >>> clf2 = RandomForestClassifier(random_state=1)
+    >>> clfm = GaussianNB()
+    >>> X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]])
+    >>> y = np.array([1, 1, 1, 2, 2, 2])
+    >>> eclf1 = StackingClassifier(estimators=[
+    ...         ('lr', clf1), ('rf', clf2)], meta_estimator=clfm, cv=2)
+    >>> eclf1 = eclf1.fit(X, y)
+    >>> print(eclf1.predict(X))
+    [1 1 1 2 2 2]
+    >>>
+    """
+
+    def __init__(self, estimators, meta_estimator, method='predict_proba',
+                 cv=None, n_jobs=1, verbose=0):
+        self.estimators = estimators
+        self.named_estimators = dict(estimators)
+        self.meta_estimator = meta_estimator
+        self.cv = cv
+        self.method = method
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+
+    def fit(self, X, y, **kwargs):
+        """ Fit the estimators.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        **kwargs: optional fit parameters
+
+
+        Returns
+        -------
+        self : object
+        """
+        if isinstance(y, np.ndarray) and len(y.shape) > 1 and y.shape[1] > 1:
+            raise NotImplementedError('Multilabel and multi-output'
+                                      ' classification is not supported.')
+
+        if self.estimators is None or len(self.estimators) == 0:
```

Comment:
How about ``if not self.estimators``? Also, please add a ``got {}`` to the error.